var/spack/repos/builtin/packages/wannier90/package.py

```diff
@@ -61,10 +89,81 @@ def edit(self, spec, prefix):
         for key, value in substitutions.items():
             filter_file(key, value, self.makefile_name)
 
-    def install(self, spec, prefix):
+        if '+shared' in self.spec:
+            if self.spec.satisfies('@:2'):
+                # this is to build a .shared wannier90 library
+                if self.spec.satisfies('platform=darwin'):
+                    filter_file('LIBRARY = ../../libwannier.a',
+                                'LIBRARY = ../../libwannier.dylib',
+                                join_path(self.stage.source_path,
+                                          'src/Makefile.2'))
+                else:
+                    filter_file('LIBRARY = ../../libwannier.a',
+                                'LIBRARY = ../../libwannier.so',
+                                join_path(self.stage.source_path,
+                                          'src/Makefile.2'))
+                objs_post = ['parameters.o', 'kmesh.o', 'io.o', 'comms.o',
+                             'utility.o', 'get_oper.o', 'constants.o',
+                             'postw90_common.o', 'wan_ham.o', 'spin.o',
+                             'dos.o', 'berry.o', 'kpath.o', 'kslice.o',
+                             'boltzwann.o', 'geninterp.o']
+                new_objs_post = ['comms.o', 'get_oper.o', 'postw90_common.o',
+                                 'wan_ham.o', 'spin.o', 'dos.o', 'berry.o',
+                                 'kpath.o', 'kslice.o', 'boltzwann.o',
+                                 'geninterp.o']
+                filter_file(''.join(['OBJS_POST  = ', ' '.join(objs_post)]),
+                            ''.join(['OBJS_POST  = ',
+                                     ' '.join(new_objs_post)]),
```

Comment:
You should be able to remove these joins too

Reply:
Thanks (and thank goodness!)